snakemake_executor_plugin_slurm/__init__.py

```diff
@@ -69,6 +69,7 @@ def run_job(self, job: JobExecutorInterface):
 
         try:
             wildcard_str = f"_{'_'.join(job.wildcards)}" if job.wildcards else ""
+            wildcard_str = wildcard_str.replace("/", ".")
```

Comment:
Why not remove it?
```suggestion
            wildcard_str = wildcard_str.replace("/", "")
```

Also, should we try to deal with other reserved characters?

Reply:
I'm not sure about the purpose. Is it about removing the '/'? Or `os.path.sep`? In that case, we should be agnostic.

Also, I think we should get rid of the try/except clause, for a simple test will suffice. I will ask about this in the issue and act accordingly.

Reply:
In my case the problem was that, since the `wildcards` contained a '/', `slurm` failed (I guess since the folder did not exist).
Not sure but, if there are other reserved characters in the `wildcards`, it might be also good to remove them.

Reply:
This is weird, because we do a makedirs. So the number of subdirs here should be no problem. Maybe rather a latency issue?

Reply:
I think the problem is that, when creating that folder, the `%s` is not replaced yet. It is `slurm` that replaces it.:
```
slurm_logfile = ".snakemake/slurm_logs/rule_test/%j_wc1_wc21/wc22_wc3.log"
```
So `snakemake` will create the folder `.snakemake/slurm_logs/rule_test/%j_wc1_wc21` and slurm will try to write on `.snakemake/slurm_logs/rule_test/123456789_wc1_wc21/wc22_wc3.log`, no?

Reply:
Exactly as @fgvieira stated -- the %j wildcard with which Slurm injects the jobid is not known when we make directories, so it cannot be used as in a parent directory for the final logfile.

In terms of reserved characters, I think we only need to worry about os.path.sep. Since wildcards themselves are extracted from other paths, I would expect them not to contain any other problematic wildcards.

Reply:
@fgvieira ah, thanks for the elaboration. I think we need a pre-configured issue report here, too. Those minimal examples really help a lot!

Reply:
@cmeesters any suggestions?

Reply:
I think the best approach is to extend the testcases that we already have, so that all executors benefit. Besides, I think I have fixed the issue in PR https://github.com/snakemake/snakemake-executor-plugin-slurm/pull/67 in a different way. The %j is now in the filename, so that we can safely create the dir before.

Reply:
And https://github.com/snakemake/snakemake-executor-plugin-slurm/pull/69 for a first try with that updated testcase.

Reply:
Seems to work well. Once https://github.com/snakemake/snakemake/pull/2810 is merged, this will be tested automatically.

Reply:
Am I right that this PR is then not necessary anymore?

Reply:
Honestly, I was checking on this PR. Yes, we can close in in favour of #67